code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java

```diff
@@ -222,10 +222,9 @@ public boolean findAndSetJWT() {
 
     @Override
     public boolean isAllowed(List<String> idList, Set<Identity> identities) {
-        boolean hasAccessToAProject = authDao.hasAccessToAnyProject(identities, false, null);
         switch (accessMode()) {
             case "restricted":
-                if (hasAccessToAProject || isWhitelisted(idList)) {
+                if (isWhitelisted(idList)) {
```

Comment:
is whitelisted implies that the user has an access to any project? or should we do

```
 hasAccessToAProject && isWhitelisted(idList)
```

meaning that the user has to be whitelisted and has an access to at least one project in order to login

instead?